Sources/AppcuesKit/Presentation/Traits/Appcues/AppcuesTargetElementTrait.swift

```diff
@@ -95,21 +94,8 @@ private extension UIView {
         }
 
         // weight the selector property matches by how distinct they are considered
-        let weightedMatches = views.map { view -> (UIView, Int) in
-            var weight = 0
-
-            if view.isMatch(for: target, on: \.accessibilityIdentifier) {
-                weight += 10_000
-            }
-
-            if view.isMatch(for: target, on: \.tag) {
-                weight += 1_000
-            }
-
-            if view.isMatch(for: target, on: \.description) {
-                weight += 100
-            }
-
+        let weightedMatches = views.map { view -> (AppcuesViewElement, Int) in
+            let weight = view.selector?.evaluateMatch(for: selector) ?? 0
```

Comment:
in theory, `strategy.findMatches(...)` could return some structure that included the weights as well, since it is checking it already to find matches. I kept the interface simpler to start and just returned the views and then have to re-check the weights here to find the best match, in this internal implementation detail.